src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.js

```diff
@@ -188,23 +178,149 @@ function discoverController(
   $scope,
   $timeout,
   $window,
-  AppState,
   Promise,
   config,
   kbnUrl,
   localStorage,
-  uiCapabilities,
-  getAppState,
-  globalState
+  uiCapabilities
 ) {
-  const filterStateManager = new FilterStateManager(globalState, getAppState, filterManager);
+  const subscriptions = new Subscription();
+  const $fetchObservable = new Subject();
+  let inspectorRequest;
+  const savedSearch = $route.current.locals.savedObjects.savedSearch;
+  if ($route.current.locals.savedObjects.savedQuery) {
+    $scope.savedQuery = $route.current.locals.savedObjects.savedQuery;
+  }
+
+  $scope.searchSource = savedSearch.searchSource;
+  $scope.indexPattern = resolveIndexPatternLoading();
+  $scope.timeRangeObj = {
+    ..._.cloneDeep(timefilter.getTime()),
+  };
+  $scope.refreshInterval = timefilter.getRefreshInterval();
+
+  const {
+    appStateContainer,
+    stopSync: stopStateSync,
+    setAppState,
+    getAppFilters,
+    resetInitialAppState,
+    replaceUrlState,
+    isAppStateDirty,
+    kbnUrlStateStorage,
+    getPreviousAppState,
+  } = getState({
+    defaultAppState: getStateDefaults(),
+    storeInSessionStorage: false,
+  });
+  if (appStateContainer.getState().index !== $scope.indexPattern.id) {
+    //used index pattern is different than the given by url/state which is invalid
+    setAppState({ index: $scope.indexPattern.id });
+  }
+  $scope.state = { ...appStateContainer.getState() };
+
+  const updateStateFromSavedQuery = savedQuery => {
+    setAppState({
+      savedQuery: savedQuery.id,
+      query: savedQuery.attributes.query,
+      filters: savedQuery.attributes.filters || [],
+    });
+
+    if (savedQuery.attributes.timefilter) {
+      timefilter.setTime({
+        from: savedQuery.attributes.timefilter.from,
+        to: savedQuery.attributes.timefilter.to,
+      });
+      if (savedQuery.attributes.timefilter.refreshInterval) {
+        timefilter.setRefreshInterval(savedQuery.attributes.timefilter.refreshInterval);
+      }
+    }
+  };
+
+  const stopSyncingAppFilters = syncAppFilters(filterManager, {
+    set: filters => setAppState({ filters: filters }),
```

Comment:
Could be simplified:
```suggestion
    set: filters => setAppState({ filters }),
```

Reply:
Thx, much prettier!